sympy/physics/control/lti.py

```diff
@@ -1144,3 +1179,210 @@ def _eval_rewrite_as_TransferFunction(self, num, den, **kwargs):
 
     def __neg__(self):
         return Feedback(-self.num, self.den)
+
+
+class TransferFunctionMatrix(Basic):
+
+    def __new__(cls, arg, shape=None, var=None):
+        if not (isinstance(arg, (list, tuple, Matrix, ImmutableMatrix)) and
```

Comment:
`SparseMatrix` is forbidden?

Reply:
```suggestion
        if not (isinstance(arg, (list, tuple, DenseMatrix)) and
```
`sympy.matrices.dense.DenseMatrix` is the common base class I think

Reply:
No, MatrixBase is the most common class

Reply:
Depends if omitting `SparseMatrix` was deliberate.

Reply:
Ok, I will allow the most common class, i.e. `MatrixBase`.